domain/application/state/charm_test.go

```diff
@@ -52,12 +52,42 @@ func (s *charmStateSuite) TestGetCharmIDByRevision(c *gc.C) {
 	})
 	c.Assert(err, jc.ErrorIsNil)
 
-	charmID, err := st.GetCharmIDByRevision(context.Background(), "foo", 1)
+	charmID, err := st.GetCharmID(context.Background(), "foo", 1, charm.CharmHubSource) // default source
 	c.Assert(err, jc.ErrorIsNil)
 	c.Check(charmID, gc.Equals, id)
 }
 
-func (s *charmStateSuite) TestSetCharmGetCharmIDByRevision(c *gc.C) {
+func (s *charmStateSuite) TestGetCharmIDLocalCharm(c *gc.C) {
+	st := NewCharmState(s.TxnRunnerFactory())
+
+	id := charmtesting.GenCharmID(c)
+
+	err := s.TxnRunner().StdTxn(context.Background(), func(ctx context.Context, tx *sql.Tx) error {
+		_, err := tx.ExecContext(ctx, `INSERT INTO charm (uuid) VALUES (?)`, id.String())
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		_, err = tx.ExecContext(ctx, `INSERT INTO charm_metadata (charm_uuid, name) VALUES (?, 'foo')`, id.String())
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		_, err = tx.ExecContext(ctx, `INSERT INTO charm_origin (reference_name, charm_uuid, revision, source_id) VALUES (?, ?, 1, 0)`, "foo", id.String())
+		if err != nil {
+			return errors.Trace(err)
+		}
+
+		return nil
+	})
```

Comment:
Is there really no method on state which does the equivalent?

Reply:
This is the test for the state package, so the ideal is to manually insert data on the db when testing the retrieval (and vice-versa) to make sure the methods are actually working, since we are already testing against the db anyways.